src/step/releases/workspace.rs

```diff
@@ -0,0 +1,64 @@
+use std::{path::Path, str::FromStr};
+
+use enum_iterator::{all, Sequence};
+
+use crate::{
+    fs::read_to_string,
+    step::releases::{
+        cargo::CargoPackage,
+        versioned_file::{PackageFormat, VersionedFile},
+        Package,
+    },
+};
+
+#[derive(Clone, Copy, Debug, Sequence)]
+enum WorkspaceFile {
+    CargoToml,
+}
+
+pub(crate) fn check_for_workspaces() -> Vec<Package> {
+    all::<WorkspaceFile>()
+        .filter_map(WorkspaceFile::get_packages)
+        .flatten()
+        .collect()
+}
+
+impl WorkspaceFile {
+    fn get_packages(self) -> Option<Vec<Package>> {
+        match self {
+            Self::CargoToml => cargo_workspace_members(Path::new("Cargo.toml")),
+        }
+    }
+}
+
+fn cargo_workspace_members(path: &Path) -> Option<Vec<Package>> {
+    let contents = read_to_string(path).ok()?;
+    let cargo_toml = toml::Value::from_str(&contents).ok()?;
+    let workspace_path = path.parent()?;
+    Some(
+        cargo_toml
+            .get("workspace")?
+            .as_table()?
+            .get("members")?
+            .as_array()?
+            .iter()
+            .filter_map(|member| member.as_str())
```

Comment:
Literally only because `.as_str()` returns an `Option`... it _should_ never return `None` unless someone has something other than a string in the members array. We could error instead of ignoring those cases though...

Reply:
I think an error would indeed make more sense. I mean a number makes no sense in that position :p